### PR TITLE
Voltage init server - simple integration

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-parent-ws</artifactId>
+        <version>10</version>
+    </parent>
+
+
+    <artifactId>gridsuite-voltage-init-server</artifactId>
```
Comment (FranckLecuyer) [inline pom.xml:20]: Missing \<groupId\> ?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+    <licenses>
+        <license>
+            <name>Mozilla Public License, Version 2.0</name>
+            <url>https://www.mozilla.org/en-US/MPL/2.0/</url>
+        </license>
+    </licenses>
+
+    <scm>
+        <connection>scm:git:https://github.com/gridsuite/voltage-init-store.git</connection>
```
Comment (FranckLecuyer) [inline pom.xml:36]: voltage-init-store ---> voltage-init-server

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java:2]: 2023

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@Configuration
+public class SwaggerConfig {
+
+    @Bean
+    public OpenAPI openAPI() {
+        return new OpenAPI()
+                .info(new Info()
+                        .title("User Admin API")
+                        .description("This is the documentation of the user admin REST API")
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java:25]: This is the documentation of the voltage init REST API

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+@Configuration
+public class SwaggerConfig {
+
+    @Bean
+    public OpenAPI openAPI() {
+        return new OpenAPI()
+                .info(new Info()
+                        .title("User Admin API")
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java:24]: Voltage init API

File: src/test/resources/logback-test.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    -->
+    <logger name="io.netty.channel.AbstractChannelHandlerContext" level="error"/>
+
+
+    <!--
+        suppress unreadable huge logs..
+    -->
+    <logger name="org.springframework.test" level="warn"/>
+    <logger name="org.springframework.boot.test" level="warn"/>
+    <logger name="org.gridsuite.shortcircuit.server" level="trace"/>
```
Comment (FranckLecuyer) [inline src/test/resources/logback-test.xml:42]: org.gridsuite.voltageinit.server

File: src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java:2]: 2023

File: src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java:2]: 2023

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    @Operation(summary = "Delete a voltage init result from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The voltage init result has been deleted")})
+    public ResponseEntity<Void> deleteResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
+        voltageInitService.deleteResult(resultUuid);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping(value = "/results", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Delete all short circuit analysis results from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "All short circuit analysis results have been deleted")})
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java:73]: All voltage init results have been deleted

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+
+        LOGGER.info("Run voltage init...");
+        Network network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids(), context.getVariantId());
+
+        CompletableFuture<OpenReacResult> future = runVoltageInitAsync(context, network, resultUuid);
+
+        return future == null ? null : future.get();
+    }
+
+    public CompletableFuture<OpenReacResult> runVoltageInitAsync(VoltageInitRunContext context, Network network, UUID resultUuid) {
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java:109]: context is not used

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: a few minor remarks
tests: not done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java (hunk 9 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java:73]: done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    @DeleteMapping(value = "/results/{resultUuid}", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Delete a voltage init result from the database")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The voltage init result has been deleted")})
+    public ResponseEntity<Void> deleteResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
+        voltageInitService.deleteResult(resultUuid);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping(value = "/results", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Delete all short circuit analysis results from the database")
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java:72]: done

File: src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java (hunk 8 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java:2]: done

File: src/main/java/org/gridsuite/voltageinit/server/dto/VoltageInitStatus.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/dto/VoltageInitStatus.java:2]: done

File: src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java (hunk 7 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java:2]: done

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 4 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java:25]: done

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 3 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java:2]: done

File: pom.xml (hunk 2 above)
Comment (etiennehomer) [reply pom.xml:36]: done

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-parent-ws</artifactId>
+        <version>10</version>
+    </parent>
+
+
+    <artifactId>gridsuite-voltage-init-server</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <name>Voltage init server</name>
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply pom.xml:23]: done

File: pom.xml (hunk 1 above)
Comment (etiennehomer) [reply pom.xml:20]: done

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0"?>
+<!--
+    Copyright (c) 2019, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply pom.xml]: done

File: src/test/java/org/gridsuite/voltageinit/server/VoltageInitControllerTest.java [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+@AutoConfigureMockMvc
+@SpringBootTest
+@ContextHierarchy({@ContextConfiguration(classes = {VoltageInitApplication.class, TestChannelBinderConfiguration.class})})
+public class VoltageInitControllerTest {
+
+    private static final UUID NETWORK_UUID = UUID.fromString("7928181c-7977-4592-ba19-88027e4254e4");
+    private static final UUID OTHER_NETWORK_UUID = UUID.fromString("06824085-db85-4883-9458-8c5c9f1585d6");
+    private static final UUID RESULT_UUID = UUID.fromString("0c8de370-3e6c-4d72-b292-d355a97e0d5d");
+
+    private static final UUID UNKNOWN_STATUS_UUID = UUID.fromString("0a22c4e4-630c-434e-9b29-980a3fea7391");
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/test/java/org/gridsuite/voltageinit/server/VoltageInitControllerTest.java]: removed

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,250 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-parent-ws</artifactId>
+        <version>10</version>
+    </parent>
+
+
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply pom.xml]: done

File: pom.xml [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+        <license>
+            <name>Mozilla Public License, Version 2.0</name>
+            <url>https://www.mozilla.org/en-US/MPL/2.0/</url>
+        </license>
+    </licenses>
+
+    <scm>
+        <connection>scm:git:https://github.com/gridsuite/voltage-init-store.git</connection>
+        <developerConnection>scm:git:https://github.com/gridsuite/voltage-init-store.git</developerConnection>
+        <url>https://github.com/gridsuite/voltage-init-store</url>
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply pom.xml]: it's an error. Fixed

File: src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/entities/GlobalStatusEntity.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/repository/VoltageInitResultRepository.java:2]: done

File: src/main/java/org/gridsuite/voltageinit/server/service/NotificationService.java [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+    private static final String FAILED_CATEGORY_BROKER_OUTPUT = NotificationService.class.getName() + ".output-broker-messages.failed";
+
+    private static final Logger RUN_MESSAGE_LOGGER = LoggerFactory.getLogger(RUN_CATEGORY_BROKER_OUTPUT);
+    private static final Logger CANCEL_MESSAGE_LOGGER = LoggerFactory.getLogger(CANCEL_CATEGORY_BROKER_OUTPUT);
+    private static final Logger STOP_MESSAGE_LOGGER = LoggerFactory.getLogger(STOP_CATEGORY_BROKER_OUTPUT);
+    private static final Logger RESULT_MESSAGE_LOGGER = LoggerFactory.getLogger(RESULT_CATEGORY_BROKER_OUTPUT);
+    private static final Logger FAILED_MESSAGE_LOGGER = LoggerFactory.getLogger(FAILED_CATEGORY_BROKER_OUTPUT);
+
+    public static final String CANCEL_MESSAGE = "Voltage init was canceled";
+    public static final String FAIL_MESSAGE = "Voltage init analysis has failed";
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/service/NotificationService.java:38]: removed

File: src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java (hunk 10 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java:109]: Yes, but i kept it as it gonna be used in next developments. The parameters will be gotten in the context. Ok to keep it ?

File: src/main/java/org/gridsuite/voltageinit/server/VoltageInitApi.java [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.voltageinit.server;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+public final class VoltageInitApi {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageInitApi.java:12]: We have this class in every ws service. Maybe it's used by swagger-ui

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java [hunk 20]
```diff
@@ -0,0 +1,103 @@
+/**
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java:1]: done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+    @ApiResponses(value = {@ApiResponse(responseCode = "200",
+            description = "The voltage init analysis has been performed")})
+    public ResponseEntity<UUID> runAndSave(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
+                                           @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
+                                           @Parameter(description = "Other networks UUID (to merge with main one))") @RequestParam(name = "networkUuid", required = false) List<UUID> otherNetworkUuids,
+                                           @Parameter(description = "Result receiver") @RequestParam(name = "receiver", required = false) String receiver,
+                                           @Parameter(description = "reportUuid") @RequestParam(name = "reportUuid", required = false) UUID reportUuid,
+                                           @Parameter(description = "reporterId") @RequestParam(name = "reporterId", required = false) String reporterId,
+                                           @RequestBody(required = false) OpenReacParameters parameters,
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java:52]: It gonna be used  in next US. I choose to keep to be homogeneous with other computation ws

File: src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java (hunk 5 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/SwaggerConfig.java]: done

File: src/test/resources/logback-test.xml (hunk 6 above)
Comment (etiennehomer) [reply src/test/resources/logback-test.xml]: done

File: src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+import static org.gridsuite.voltageinit.server.service.NotificationService.HEADER_USER_ID;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+@RestController
+@RequestMapping(value = "/" + VoltageInitApi.API_VERSION)
+@Tag(name = "Voltage init server")
+public class VoltageIniController {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageIniController.java]: done

File: src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java (hunk 10 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java:109]: yes!

**Review by FranckLecuyer — APPROVED** [review]

**Review by dbraquart — APPROVED** [review]
code: ok
tests: not done

**Pull request merged**